### ISSUE TITLE
Amplify.addPlugin produces a crash when I start my app for API level lower than 21

### ISSUE THREAD

**Issue by Mariani88** [aws-amplify/amplify-android#657]
Hi Amplify devs, I have an issue when I start my app and add plugins. The issue occurs for Android devices with API level lower than 21. For devices with API level higher than 21, my app works correctly.

I'm using Cognito with guest authentication and S3, when I build my app, it builds successfully, but when I start it, it crashes with next error:

```console
07-20 20:47:35.760 3684-3684/com.marcast.sistmovildonatto.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.marcast.sistmovildonatto.debug, PID: 3684
    java.lang.NoClassDefFoundError: com.amazonaws.mobile.client.AWSMobileClient
        at com.amplifyframework.auth.cognito.AWSCognitoAuthPlugin.<init>(AWSCognitoAuthPlugin.java:2)
        at com.marcast.sistmovildonatto.DidonatoApplication.onCreate(DidonatoApplication.kt:16)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1007)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4344)
        at android.app.ActivityThread.access$1500(ActivityThread.java:135)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:5017)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
        at dalvik.system.NativeStart.main(Native Method)
```

I have these dependencies in my `build.gradle`:

```groovy
implementation 'com.amplifyframework:core:1.0.0'
implementation 'com.amplifyframework:aws-storage-s3:1.0.0'
implementation 'com.amplifyframework:aws-auth-cognito:1.0.0'
```

My app initializes Amplify in an `Application` class like this:

```kotlin
class DidonatoApplication: Application() {
    override fun onCreate() {
        super.onCreate()

        try {
            Amplify.addPlugin(AWSCognitoAuthPlugin())
            Amplify.addPlugin(AWSS3StoragePlugin())
            Amplify.configure(applicationContext)
            Log.i("DidonatoMobile", "Initialized Amplify")
        } catch (error: AmplifyException) {
            Log.e("DidonatoMobile", "Could not initialize Amplify", error)
        }
    }
}
```

The app crash occurs at the line `Amplify.addPlugin(AWSCognitoAuthPlugin())`.

Thank you very much! I will wait for your response.

**Comment by rjuliano** [comment]
As a first step in troubleshooting this, could you please try to update your Amplify dependencies to 1.1.1 and see if the error persists? I'm running your code locally with against API 16 and I'm not seeing the error.

**Comment by Mariani88** [comment]
Hi! I just have updated dependencies to 1.1.1, but the error still persists. Do you need to see my aws_configuration.json?

**Comment by Mariani88** [comment]
Hi! any news about this? I have been reviewing my code it never works for Android 4.

**Comment by rjuliano** [comment]
Which API level? I've tested with [API level 16 (4.1.x - Jelly Bean)](https://source.android.com/setup/start/build-numbers) and higher which is what [we currently support](https://docs.amplify.aws/lib/project-setup/prereq/q/platform/android).

The only other possible cause for the `java.lang.NoClassDefFoundError: com.amazonaws.mobile.client.AWSMobileClient` during initialization is if the amplifyframework.json config file cannot be found under the `<app>/src/res/raw` folder. But if it works for API level 21+ as you mentioned, then this is probably not the cause.

**Comment by Mariani88** [comment]
I have tested my app with all API levels between 16 and 19. It crashes into all of them. I reviewed all commits, including the first commit when I integrated the amplify and it chashed too. I have no idea what is the problem. Any suggest for this problem?

**Comment by jamesonwilliams** [comment]
Remark, for cross-reference: this issue is also documented on StackOverflow, as ["NoClassDefFoundError: com/amazonaws/mobile/client/AWSMobileClient"](https://stackoverflow.com/q/63145280/695787).

**Comment by jpignata** [comment]
@Mariani88 This should be fixed in 1.2.0. Please comment here if that's not the case. Thank you so much for the report!
